src/frontend/src/app/asset-readings/chart-modal/chart-modal.component.ts

```diff
@@ -36,26 +38,41 @@ export class ChartModalComponent implements OnInit {
     }
     chart_modal.classList.remove('is-active');
     this.assetCode = '';
-    if (this.isValidData) {
+    if (this.isValidData && this.isReadingsAvailable) {
       this.chartComp.ngOnDestroy();
     }
   }
 
-  public plotReadingsGraph(assetCode) {
+  public plotReadingsGraph(assetCode, limit = 0, offset = 0) {
     this.isValidData = true;
+    this.isReadingsAvailable = true;
+    this.isInvalidInput = false;
+    if (limit == null) {
+      limit = 0;
+    }
+    if (offset == null) {
+      offset = 0;
+    }
+    if (limit < 0 || limit > 1000 || offset < 0 || !Number.isInteger(limit)
```

Comment:
separate limit and offset check!
`if (limit < 0 || limit > 1000 || !Number.isInteger(limit)`
and next if for offset